### ISSUE TITLE
write some docs on writing tests

### ISSUE THREAD

**Issue by shiftkey** [desktop/desktop#5496]
An incomplete list of things to cover:

 - unit tests versus integration tests
 - how tests are organized
 - how Git tests work
 - adding Git repository fixtures
 - how to generate coverage data for tests

**Comment by outofambit** [comment]
YES 💯

thank you for capturing this in an issue!

**Comment by shiftkey** [comment]
👍 there's some things I've been thinking about around, but I've got some other things to ship soon so feel free to make a start

**Comment by KennethSweezy** [comment]
Is there a particular reason this issue is still open? If it has been completed shouldn't it be closed?

**Comment by shiftkey** [comment]
@KennethSweezy I don't believe we've resolved some of the tasks in the list on that first comment
